### PR TITLE
[interop] restructure Swift -> C++ interop tests

### PR DESCRIPTION
**Pull request by hyp**
We will place tests that test the semantics and runtime of the generated
C++ code into Interop/SwiftToCxx. PrintCxx test directory will only be
used to test high level flow without concerning itself about the C++ semantics.

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please smoke test

**Comment by hyp** [comment]
@swift-ci please smoke test

**Comment by hyp** [comment]
@swift-ci please test

**Review by zoecarver — APPROVED** [review]

File: test/lit.cfg [hunk 1]
```diff
@@ -558,13 +558,39 @@ config.substitutions.append( ('%clang_apinotes',
...
 # Note: %clang is the locally-built clang.
 # To get Xcode's clang, use %target-clang.
 config.substitutions.append( ('%clang',
                               "%r %s" %
                                 (config.clang, clang_mcp_opt)) )
 
+# Verifies that a C++ file can be compiled without warnings except for some exceptions.
+config.substitutions.insert(0, ('%check-cxx-header-in-clang',
+  '%%clang++ -fsyntax-only -x c++-header '
+  # Turn on all warnings and promote them to errors
```
Comment (zoecarver) [inline test/lit.cfg:575]: Thank you for all the comments here :)

**Comment by hyp** [comment]
@swift-ci please test

**Pull request merged**